pkgs/development/ocaml-modules/plotkicadsch/open-process.patch

```diff
@@ -0,0 +1,19 @@
+diff --git a/plotkicadsch/src/sysAbst.ml b/plotkicadsch/src/sysAbst.ml
+index c3c7b52..4300e8f 100644
+--- a/plotkicadsch/src/sysAbst.ml
++++ b/plotkicadsch/src/sysAbst.ml
+@@ -30,9 +30,9 @@ let detect_os () : os =
+   if Sys.win32 then Windows
+   else if Sys.cygwin then Cygwin
+   else
+-    let ((in_ch, _, _) as uname) = UnixLabels.open_process_full "uname" ~env:[| |] in
++    let ((in_ch, _) as uname) = UnixLabels.open_process "uname" in
+     let os = input_line in_ch in
+-    ignore (UnixLabels.close_process_full uname) ;
++    ignore (UnixLabels.close_process uname) ;
```

Comment:
I noticed that on NixOS, the upstream code does not work whether I am installing this package via opam or Nix.

I'm not sure if it's actually safe to do this though.

Reply:
Did you report your issue upstream? They might be able to tell you whether your patch is ok.

Reply:
Per your suggestion I have asked the maintainer at https://github.com/jnavila/plotkicadsch/issues/68.